### PR TITLE
Local database fp 63

### PR DESCRIPTION
**Pull request by JosephOri**
## Task link from JIRA:
link:  https://mssplinter10-1711638164183.atlassian.net/jira/software/projects/FP/boards/6?selectedIssue=FP-63

## Description:
added migrations and modified the backend in order to run with postgres dockerhub image

### CODE REVIEW

File: libs/common/src/config/database.ts
Comment (nitzanto) [inline libs/common/src/config/database.ts:1]: You have duplicated code here and in data-source.ts

File: libs/common/src/data-source/data-source.ts
Comment (nitzanto) [inline libs/common/src/data-source/data-source.ts:1]: duplicated code. Why use dotenv.config()? You can use NestJS's config services.

File: libs/common/src/entities/chat.entity.ts
Comment (nitzanto) [inline libs/common/src/entities/chat.entity.ts:1]: Why did you move all of the entities into the common folder? They're not common.

File: libs/common/src/entities/index.ts
Comment (nitzanto) [inline libs/common/src/entities/index.ts:1]: Should not be here.

File: libs/common/src/migrations/1723712816562-InitialMigration.ts
Comment (nitzanto) [inline libs/common/src/migrations/1723712816562-InitialMigration.ts:1]: All of the migrations shouldnt be in one migration file. But rather should be small and focused migrations in each file.

When you'd want to change one of the migrations it'll affect the other tables as well. Also you're mixing logic here.

**Review by nitzanto — CHANGES_REQUESTED** [review]
Please read the comments.

File: libs/common/src/entities/chat.entity.ts
Comment (JosephOri) [reply libs/common/src/entities/chat.entity.ts:1]: I need all of my entities for the migrations

**Review by JosephOri — COMMENTED** [review]
a

File: libs/common/src/entities/index.ts
Comment (JosephOri) [reply libs/common/src/entities/index.ts:1]: it should

File: libs/common/src/migrations/1723712816562-InitialMigration.ts
Comment (JosephOri) [reply libs/common/src/migrations/1723712816562-InitialMigration.ts:1]: WTF???

File: libs/common/src/entities/chat.entity.ts
Comment (nitzanto) [reply libs/common/src/entities/chat.entity.ts:1]: They're not common. Should not be here. There are other ways to retrieve the entitles, read the documentation or google it.

File: libs/common/src/entities/index.ts
Comment (nitzanto) [reply libs/common/src/entities/index.ts:1]: Common folder is intended for reuseable code across the application that should be shared within more than one resource. Please explain how this is the case here

File: libs/common/src/migrations/1723712816562-InitialMigration.ts
Comment (nitzanto) [reply libs/common/src/migrations/1723712816562-InitialMigration.ts:1]: This is a bad practice. Please re-evaluate your code here and read my comment again.

**Review by nitzanto — APPROVED** [review]
Great
